modules/widgets/recurring-payments.php

```diff
@@ -0,0 +1,193 @@
+<?php // phpcs:disable WordPress.Files.FileName.InvalidClassFileName
+/**
+ * Disable direct access/execution to/of the widget code.
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( 'Jetpack_Recurring_Payments_Widget' ) ) {
+	/**
+	 * Recurring Payments Button
+	 *
+	 * Display a Recurring Payments Button as a Widget.
+	 */
+	class Jetpack_Recurring_Payments_Widget extends WP_Widget {
+		/**
+		 * Constructor.
+		 */
+		public function __construct() {
+			parent::__construct(
+				'jetpack_recurring_payments_widget',
+				/** This filter is documented in modules/widgets/facebook-likebox.php */
+				apply_filters( 'jetpack_widget_name', __( 'Recurring Payments', 'jetpack' ) ),
+				array(
+					'classname'                   => 'jetpack-recurring-payments',
+					'description'                 => __( 'Add a Recurring Payments Button as a Widget.', 'jetpack' ),
+					'customize_selective_refresh' => true,
+				)
+			);
+
+		}
+
+		/**
+		 * Return an associative array of default values.
+		 * These values are used in new widgets.
+		 *
+		 * @param int $first_product_id - ID of the first product post.
+		 * @see self::get_first_product_id.
+		 *
+		 * @return array Default values for the widget options.
+		 */
+		private function defaults( $first_product_id ) {
+			return array(
+				'title'           => '',
+				'product_post_id' => $first_product_id,
+				'text'            => __( 'Contribution', 'jetpack' ),
+			);
+		}
+
+		/**
+		 * Get the first product to serve as a default.
+		 *
+		 * @return int - the id of the first product.
+		 */
+		private function get_first_product_id() {
+			$first_product = get_posts(
+				array(
+					'numberposts' => 1,
+					'orderby'     => 'date',
+					'post_type'   => Jetpack_Memberships::$post_type_plan,
+					'post_status' => 'publish',
+				)
+			);
+			if ( ! $first_product || is_wp_error( $first_product ) ) {
+				return 0;
+			}
+			return $first_product->ID;
```

Comment:
I'm getting a notice here:
```
Notice: Trying to get property 'ID' of non-object in wp-content/plugins/jetpack-dev/modules/widgets/recurring-payments.php on line 68
```

`$first_product` seems to be an array containing the `WP_Post`.